### PR TITLE
Examples refactoring (Top of giant branch includes the Multi version support changes)

### PR DESCRIPTION
**Pull request by victorwiebe**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -17,38 +18,38 @@ import * as conversionUtils from './utils/convert';
...
+  //   export import VolumeRestrictionTransferManager = VolumeRestrictionTransferManagerTransactionParams;
+  //   export import RestrictedPartialSaleTransferManager = RestrictedPartialSaleTransferManagerTransactionParams;
+  //   export import VestingEscrowWallet = VestingEscrowWalletTransactionParams;
+  //   export import ERC20 = ERC20TransactionParams;
+  //   export import PolyTokenFaucet = PolyTokenFaucetTransactionParams;
+  //   export import PolyToken = PolyTokenTransactionParams;
+     export import SecurityToken = SecurityTokenTransactionParams;
+}
 
-export { conversionUtils };
```
Comment (monitz87) [inline src/index.ts]: Why was this removed?

File: src/index.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
 /* istanbul ignore file */
-import * as conversionUtils from './utils/convert';
```
Comment (monitz87) [inline src/index.ts]: Why was this removed?

File: src/index.ts [hunk 3]
```diff
@@ -17,38 +18,38 @@ import * as conversionUtils from './utils/convert';
...
-//   export import ManualApprovalTransferManager = ManualApprovalTransferManagerTransactionParams;
-//   export import PercentageTransferManager = PercentageTransferManagerTransactionParams;
-//   export import VolumeRestrictionTransferManager = VolumeRestrictionTransferManagerTransactionParams;
-//   export import VestingEscrowWallet = VestingEscrowWalletTransactionParams;
-//   export import ERC20 = ERC20TransactionParams;
-//   export import PolyTokenFaucet = PolyTokenFaucetTransactionParams;
-//   export import PolyToken = PolyTokenTransactionParams;
-//   export import SecurityToken = SecurityTokenTransactionParams;
-// }
+export namespace TransactionParams {
```
Comment (monitz87) [inline src/index.ts]: Why was this uncommented?

File: src/index.ts [hunk 4]
```diff
@@ -1,13 +1,14 @@
 /* istanbul ignore file */
-import * as conversionUtils from './utils/convert';
-// import { FeatureRegistryTransactionParams } from './contract_wrappers/registries/feature_registry_wrapper';
+import { FeatureRegistryTransactionParams } from './contract_wrappers/registries/feature_registry_wrapper';
```
Comment (monitz87) [inline src/index.ts]: This should be commented

File: src/index.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+import { GeneralTransferManagerTransactionParams } from './contract_wrappers/modules/transfer_manager/general_transfer_manager_wrapper/common';
```
Comment (monitz87) [inline src/index.ts]: This should be commented

File: src/index.ts [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
+import {SecurityTokenTransactionParams} from './contract_wrappers/tokens/security_token_wrapper';
```
Comment (monitz87) [inline src/index.ts]: Separate brackets from variable name (also should be commented)

File: src/types.ts [hunk 7]
```diff
@@ -255,7 +258,9 @@ export type ContractEventArgs_3_0_0 =
   | BlacklistTransferManagerEventArgs_3_0_0
   | VestingEscrowWalletEventArgs_3_0_0;
 
-export type ContractEventArgs = ContractEventArgs_3_0_0;
+export type ContractEventArgs_3_1_0 = RestrictedPartialSaleTMEventArgs_3_1_0;
```
Comment (monitz87) [inline src/types.ts]: I'm guessing you intended to add `RestrictedPartialSaleTMEventArgs_3_1_0` to the union type but instead you replaced this

File: src/index.ts (hunk 6 above)
Comment (victorwiebe) [reply src/index.ts]: Jere, why do you have it commented?

I have them uncommented because I am using some of the transactionParams in my examples, for example using the general transfer manager to get the kyc data params structure. I can remove the interfaces from the code but let me know

File: examples/restrictedPartialSaleTransferManager.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+import { registerTicker } from './registerTicker';
+import { launchToken } from './launchToken';
+import {AddingModuleOpts, addModule, moduleInstancesLookup} from './modules';
+import { addInvestorsToWhitelist } from './addInvestorsToWhitelist';
+import { issueTokenToInvestors } from './issueTokenToInvestor';
+import {RestrictedPartialSaleTransferManagerData} from '../src/contract_wrappers/tokens/security_token_wrapper';
+
+// This file acts as a valid sandbox for using a volume restriction transfer manager module on an unlocked node (like ganache)
+
+window.addEventListener('load', async () => {
```
[the remark this answers is not in the document]
Comment (victorwiebe) [reply examples/restrictedPartialSaleTransferManager.ts]: Not ready yet

[the remark this answers is not in the document]
Comment (victorwiebe) [reply examples/restrictedPartialSaleTransferManager.ts]: This was not done for any of the modules yet. We should open a new task and branch for that as it is a lot more work.

File: src/types.ts (hunk 7 above)
Comment (victorwiebe) [reply src/types.ts]: This is the code I am using now:

```
export type ContractEventArgs_3_1_0 = RestrictedPartialSaleTMEventArgs_3_1_0;
export type ContractEventArgs = ContractEventArgs_3_0_0 | ContractEventArgs_3_1_0;
```
Let me know if thats incorrect

File: examples/restrictedPartialSaleTransferManager.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (victorwiebe) [reply examples/restrictedPartialSaleTransferManager.ts]: On slack I mentioned

> . Right now RestrictedPartialSaleTM has the sample code for a future sandbox, it uses the modular example components. Examples that are currently modular include registerTicker, launchToken, addInvestorsToWhitelist, modules, issueTokensToInvestors . All other examples are still the same, they have only been changed to compile with the refactoring.

File: src/index.ts (hunk 6 above)
Comment (monitz87) [reply src/index.ts]: I commented them because I thought we weren't gonna use them until we figure out how to make them compatible with the multi version structure we use now.

If you're using them in the examples I guess it's fine to leave them uncommented, so that when we do apply the change we'll know to adapt the examples because the editor will start complaining

File: src/types.ts (hunk 7 above)
Comment (monitz87) [reply src/types.ts]: Kinda, but it's kinda my fault as well, I forgot to declare `ContractEventArgs_3_1_0` before

`ContractEventArgs_3_1_0` should be a union type of all the event args for the contracts that are represented in `ContractEvents_3_1_0`, not just the RPSTM

But yeah, you're on the right path

Comment (victorwiebe) [reply src/types.ts]: At the time there was no other 3_1 contract event args

File: examples/restrictedPartialSaleTransferManager.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (monitz87) [reply examples/restrictedPartialSaleTransferManager.ts]: Oh, I see, my bad.

File: src/types.ts (hunk 7 above)
Comment (victorwiebe) [reply src/types.ts]: Voting will be apart of this as well
